mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c

```diff
@@ -605,29 +637,47 @@ uint8_t UiLcdHy28_SpiReadByteFast()
 void UiLcdHy28_WriteIndexSpi(unsigned char index)
 {
     UiLcdHy28_SpiLcdCsEnable();
+#ifdef USE_GFX_ILI9486
 
+    GPIO_ResetBits(LCD_RS_PIO, LCD_RS);
+    UiLcdHy28_SpiSendByte(0);
+    UiLcdHy28_SpiSendByte(index);
+
+#else
     UiLcdHy28_SpiSendByte(SPI_START | SPI_WR | SPI_INDEX);   /* Write : RS = 0, RW = 0       */
     UiLcdHy28_SpiSendByte(0);
     UiLcdHy28_SpiSendByte(index);
 
     UiLcdHy28_LcdSpiFinishTransfer();
+#endif
 }
 
 void UiLcdHy28_WriteDataSpi( unsigned short data)
 {
     UiLcdHy28_SpiLcdCsEnable();
+#ifdef USE_GFX_ILI9486
+
+    GPIO_SetBits(LCD_RS_PIO, LCD_RS);
+    UiLcdHy28_SpiSendByte((data >>   8));                    /* Write D8..D15                */
+    UiLcdHy28_SpiSendByte((data & 0xFF));                    /* Write D0..D7   */
 
+#else
```

Comment:
Move the #ifdef so that it covers the real difference (the RS bit vs. the extra SPI byte).
In fact, we could simply call UiLcdHy28_WriteDataSpiStart. Much better.

Reply:
Sorry, i don't understand

Reply:
Final code:
 UiLcdHy28_SpiLcdCsEnable();
UiLcdHy28_WriteDataSpiStart();
UiLcdHy28_SpiSendByte((data >>   8));                    /* Write D8..D15                */
UiLcdHy28_SpiSendByte((data & 0xFF));                    /* Write D0..D7   */
UiLcdHy28_LcdSpiFinishTransfer();

No need for ifdef at all.

Reply:
OK i agree, it could be even shorter - without UiLcdHy28_SpiLcdCsEnable();  it is already in UiLcdHy28_WriteDataSpiStart();